cmd/deprecated-images.go

```diff
@@ -0,0 +1,86 @@
+package cmd
+
+import (
+	"github.com/CircleCI-Public/circleci-cli/api"
+	"github.com/CircleCI-Public/circleci-cli/pipeline"
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+	"gopkg.in/yaml.v3"
+)
+
+// CircleCI Linux VM images that will be permanately removed on May 31st.
+var deprecatedImages = []string{
+	"circleci/classic:201710-01",
+	"circleci/classic:201703-01",
+	"circleci/classic:201707-01",
+	"circleci/classic:201708-01",
+	"circleci/classic:201709-01",
+	"circleci/classic:201710-02",
+	"circleci/classic:201711-01",
+	"circleci/classic",
+	"circleci/classic:latest",
+	"circleci/classic:edge",
+	"circleci/classic:201808-01",
+	"ubuntu-1604:201903-01",
+	"ubuntu-1604:202004-01",
+	"ubuntu-1604:202007-01",
+	"ubuntu-1604:202010-01",
+	"ubuntu-1604:202101-01",
+	"ubuntu-1604:202104-01",
+}
+
+// Simplified Config -> job Structure for an image
+
+type job struct {
+	Machine interface{} `yaml:"machine"`
+}
+
+// Simplified Config Structure for an image
+type processedConfig struct {
+	Jobs map[string]job `yaml:"jobs"`
+}
+
+// Processes the config down to v2.0, then checks image used against the block list
+func deprecatedImageCheck(opts configOptions, flags *pflag.FlagSet, path string) error {
+
+	var params pipeline.Parameters
+
+	orgSlug, _ := flags.GetString("org-slug")
+
+	// first process the config so we can get all images
+	response, err := api.ConfigQuery(opts.cl, path, orgSlug, params, pipeline.LocalPipelineValues())
+	if err != nil {
+		return err
+	}
+
+	aConfig := processedConfig{}
+	err = yaml.Unmarshal([]byte(response.OutputYaml), &aConfig)
+	if err != nil {
+		return err
+	}
+
+	// check each job
+	for key := range aConfig.Jobs {
+
+		switch aConfig.Jobs[key].Machine.(type) {
+		case bool, nil:
+			// using machine true
+			continue
+		}
+
+		image := aConfig.Jobs[key].Machine.(map[string]interface{})["image"]
+
+		// using the `docker`/`xcode` executors
+		if image == nil {
+			continue
+		}
+
+		for _, v := range deprecatedImages {
+			if image.(string) == v {
+				return errors.New("The config is using a deprecated Linux VM image. Please see <insert URL here>. This error can be ignored by using the '--ignore' flag.")
```

Comment:
Did you mean to replace `<insert URL here>`, or are you planning to merge with the placeholder?

Reply:
Also, I would suggest including the specific image name in this error, for clarity.

Reply:
Yes, that was on purpose. I was waiting for the Content Team to publish the blog post, which they did late last night. Now that it's public, I can put the proper URL.

Reply:
Added the image to the error message.